kyaml/yaml/rnode.go

```diff
@@ -500,11 +1105,24 @@ func (rn *RNode) GetAnnotations() map[string]string {
 	return rn.getMapFromMeta(meta, AnnotationsField)
 }
 
+// SetAnnotation tries to set an annotations using the provided key and value.
+func (rn *RNode) SetAnnotation(key, value string) error {
+	annotations := rn.GetAnnotations()
+	annotations[key] = value
+	return rn.SetAnnotations(annotations)
```

Comment:
I really like this. I've written out these three lines separately so many times, it will be great to have a method to do it for me.